heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -0,0 +1,422 @@
+            'You must either remove all "preset" values from your Jest configuration, use the "extends" ' +
+            'property, or set the "disableConfigurationModuleResolution" option to "true" on the Jest ' +
+            'plugin in heft.json'
+        );
+      }
+    }
+
+    const jestArgv: Config.Argv = {
+      watch: test.properties.watchMode,
+
+      // In debug mode, avoid forking separate processes that are difficult to debug
+      runInBand: heftSession.debugMode,
+      debug: heftSession.debugMode,
+      detectOpenHandles: !!test.properties.detectOpenHandles,
+
+      cacheDirectory: this._getJestCacheFolder(heftConfiguration),
+      updateSnapshot: test.properties.updateSnapshots,
+
+      listTests: false,
+      rootDir: buildFolder,
+
+      silent: test.properties.silent,
+      testNamePattern: test.properties.testNamePattern,
+      testPathPattern: test.properties.testPathPattern ? [...test.properties.testPathPattern] : undefined,
+      testTimeout: test.properties.testTimeout,
+      maxWorkers: test.properties.maxWorkers,
+
+      passWithNoTests: options?.passWithNoTests,
+
+      $0: process.argv0,
+      _: []
+    };
+
+    if (!test.properties.debugHeftReporter) {
+      // Extract the reporters and transform to include the Heft reporter by default
+      jestArgv.reporters = this._extractHeftJestReporters(
+        jestConfig,
+        heftSession,
+        heftConfiguration,
+        jestLogger
+      );
+    } else {
+      jestLogger.emitWarning(
+        new Error('The "--debug-heft-reporter" parameter was specified; disabling HeftJestReporter')
+      );
+    }
+
+    if (test.properties.findRelatedTests && test.properties.findRelatedTests.length > 0) {
+      // Pass test names as the command line remainder
+      jestArgv.findRelatedTests = true;
+      jestArgv._ = [...test.properties.findRelatedTests];
+    }
+
+    // Stringify the config and pass it into Jest directly
+    jestArgv.config = JSON.stringify(jestConfig);
+
+    const {
+      // Config.Argv is weakly typed.  After updating the jestArgv object, it's a good idea to inspect "globalConfig"
+      // in the debugger to validate that your changes are being applied as expected.
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      globalConfig,
+      results: jestResults
+    } = await runCLI(jestArgv, [buildFolder]);
+
+    if (jestResults.numFailedTests > 0) {
+      jestLogger.emitError(
+        new Error(
+          `${jestResults.numFailedTests} Jest test${jestResults.numFailedTests > 1 ? 's' : ''} failed`
+        )
+      );
+    } else if (jestResults.numFailedTestSuites > 0) {
+      jestLogger.emitError(
+        new Error(
+          `${jestResults.numFailedTestSuites} Jest test suite${
+            jestResults.numFailedTestSuites > 1 ? 's' : ''
+          } failed`
+        )
+      );
+    }
+  }
+
+  private async _validateJestTypeScriptDataFileIfExistsAsync(buildFolder: string): Promise<void> {
```

Comment:
Seems like this would naturally exist as a member of the `JestTypeScriptDataFile` class.